shell/platform/darwin/macos/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -520,16 +627,22 @@ - (void)setMarkedText:(id)string
   if (_activeModel == nullptr) {
     return;
   }
+  std::string textBeforeChange = _activeModel->GetText().c_str();
   if (!_activeModel->composing()) {
     _activeModel->BeginComposing();
   }
 
   // Input string may be NSString or NSAttributedString.
-  BOOL isAttributedString = [string isKindOfClass:[NSAttributedString class]];
-  NSString* marked_text = isAttributedString ? [string string] : string;
-  _activeModel->UpdateComposingText([marked_text UTF8String]);
+  std::string marked_text = [string UTF8String];
```

Comment:
According to the comment `string` here can be a `NSAttributedString` which is not a subtype of `NSString` and does not respond to the `UTF8String` selector I think, so it will probably crash the app if a `NSAttributedString` is passed to this method.